src/software/world/team_test.cpp

```diff
@@ -551,3 +551,44 @@ TEST_F(TeamTest, update_timestamp_history_from_team_update)
     EXPECT_EQ(three_seconds_future, team.getTimestampHistory()[0]);
     EXPECT_EQ(4, team.getTimestampHistory().size());
 }
+
+TEST_F(TeamTest, get_all_robots_except_goalie_in_team_of_3)
+{
+    Team team = Team(Duration::fromMilliseconds(2000));
+
+    Robot robot_0 = Robot(0, Point(0, 1), Vector(-1, -2), Angle::half(),
+                          AngularVelocity::threeQuarter(), current_time);
+
+    Robot robot_1 = Robot(1, Point(3, -1), Vector(), Angle::zero(),
+                          AngularVelocity::zero(), current_time);
+
+    Robot robot_2 = Robot(2, Point(3, -1), Vector(), Angle::zero(),
+                          AngularVelocity::zero(), current_time);
+
+    team.updateRobots({robot_0, robot_1, robot_2});
+    team.assignGoalie(0);
+
+    EXPECT_EQ(std::vector<Robot>({robot_1, robot_2}), team.getAllRobotsExceptGoalie());
```

Comment:
The javadoc comment on `Team::getAllRobotsExceptGoalie` provides no guarantee that the vector of `Robot`s is in order. We should be writing tests that only test that the function does what it says, and not any other behaviour, such as the ordering of the returned `Robot`s. In this case, we should be specifically checking that the goalie is _not_ in this vector, and checking that the vector contains all the other robots. 

We can do this like the following:
```
std::unordered_set<int> returned_robot_ids;
for (const auto& robot : team.getAllRobotsExceptGoalie()) {
    returned_robot_ids.emplace(robot.id());
}
EXPECT_EQ(returned_robot_ids.find(0), returned_robot_ids.end());
EXPECT_NE(returned_robot_ids.find(1), returned_robot_ids.end());
EXPECT_NE(returned_robot_ids.find(2), returned_robot_ids.end());
```
`unordered_set::find()` will return the `end` iterator if it cannot find the given item in the set; otherwise, it will return an iterator pointing to the found item.

Reply:
When i imported the above code into the test, it gave me an error saying "no member named unordered_set in namespace std" with the code highlighted in red. Just wondering how I should format this in the test so it is accepted?

Reply:
This just means you need to include unordered_set from the c++ std library. So at the top of your test file you can include it by doing "#include <unordered_set>"

Reply:
changed!